### PR TITLE
CSI-3254 add svc snapshots for hyperswap

### PR DESCRIPTION
**Pull request by oriyarde**

### CODE REVIEW

File: controller/tests/array_action/svc/array_mediator_svc_test.py [hunk 1]
```diff
@@ -406,20 +411,43 @@ def test_get_object_by_id_volume_success(self):
...
+        vols_to_return = [source_vol_to_copy_from, source_vol_to_copy_from]
+
+        if different_pool_site:
+            pools_to_return = [Munch({'site_name': 'pool_site'}),
+                               Munch({'site_name': 'source_volume_site'}),
+                               Munch({'site_name': 'other_volume_site'}),
+                               Munch({'site_name': 'pool_site'})]
+            self.svc.client.svcinfo.lsmdiskgrp.side_effect = self._mock_cli_objects(pools_to_return)
+
+            aux_vols = [self._get_cli_volume(name='other_volume', pool_name='other_volume_pool'),
```
Comment (ArbelNathan) [inline controller/tests/array_action/svc/array_mediator_svc_test.py:436]: please dont abbreviate, `auxiliary_volumes`. (and any other `vol` in the method)

File: controller/tests/array_action/svc/array_mediator_svc_test.py [hunk 2]
```diff
@@ -406,20 +411,43 @@ def test_get_object_by_id_volume_success(self):
...
 
-    def _prepare_mocks_for_create_snapshot(self, support_deduplicated_copy=True, source_has_deduplicated_copy=False):
+    def _get_custom_dedup_cli_volume(self, support_deduplicated_copy, with_deduplicated_copy, name='source_volume',
+                                     pool_name='pool_name'):
+        volume = self._get_cli_volume(with_deduplicated_copy, name=name, pool_name=pool_name)
+        if not support_deduplicated_copy:
+            del volume.deduplicated_copy
+        return volume
+
+    def _prepare_mocks_for_create_snapshot(self, support_deduplicated_copy=True, source_has_deduplicated_copy=False,
```
Comment (ArbelNathan) [inline controller/tests/array_action/svc/array_mediator_svc_test.py:421]: are the part you added are relevant for all (or at least "most") create_snapshot tests?

File: controller/tests/array_action/svc/array_mediator_svc_test.py (hunk 1 above)
Comment (oriyarde) [reply controller/tests/array_action/svc/array_mediator_svc_test.py]: done

Comment (ArbelNathan) [reply controller/tests/array_action/svc/array_mediator_svc_test.py]: tnx

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Pull request merged**